cmd/recorder/utils.go

```diff
@@ -158,3 +161,97 @@ func (ct *clientTransport) RoundTrip(req *http.Request) (*http.Response, error)
+	// it to the list of allowed insecure origins. We assume this would only
+	// happen either on internal private networks or for development/testing purposes.
+	if u.Scheme == "http" {
+		insecureOrigins = []string{
+			siteURL,
+		}
+	}
+
+	if devMode := os.Getenv("DEV_MODE"); devMode == "true" {
+		insecureOrigins = append(insecureOrigins, []string{
+			"http://172.17.0.1:8065",
+			"http://host.docker.internal:8065",
+			"http://mm-server:8065",
+			"http://host.minikube.internal:8065",
+		}...)
+	}
+
+	return insecureOrigins, nil
+}
+
+func genChromiumOptions(cfg config.RecorderConfig) ([]chromedp.ExecAllocatorOption, []chromedp.ContextOption, error) {
+	opts := []chromedp.ExecAllocatorOption{
+		chromedp.NoFirstRun,
+		chromedp.NoDefaultBrowserCheck,
+		chromedp.DisableGPU,
+
+		// puppeteer default behavior
+		chromedp.Flag("disable-infobars", true),
+		chromedp.Flag("disable-background-networking", true),
+		chromedp.Flag("enable-features", "NetworkService,NetworkServiceInProcess"),
+		chromedp.Flag("disable-background-timer-throttling", true),
+		chromedp.Flag("disable-backgrounding-occluded-windows", true),
+		chromedp.Flag("disable-breakpad", true),
+		chromedp.Flag("disable-client-side-phishing-detection", true),
+		chromedp.Flag("disable-default-apps", true),
+		chromedp.Flag("disable-dev-shm-usage", true),
+		chromedp.Flag("disable-extensions", true),
+		chromedp.Flag("disable-features", "site-per-process,TranslateUI,BlinkGenPropertyTrees"),
+		chromedp.Flag("disable-hang-monitor", true),
+		chromedp.Flag("disable-ipc-flooding-protection", true),
+		chromedp.Flag("disable-popup-blocking", true),
+		chromedp.Flag("disable-prompt-on-repost", true),
+		chromedp.Flag("disable-renderer-backgrounding", true),
+		chromedp.Flag("disable-sync", true),
+		chromedp.Flag("force-color-profile", "srgb"),
+		chromedp.Flag("metrics-recording-only", true),
+		chromedp.Flag("safebrowsing-disable-auto-update", true),
+		chromedp.Flag("password-store", "basic"),
+		chromedp.Flag("use-mock-keychain", true),
+		chromedp.Flag("use-fake-ui-for-media-stream", true),
+		chromedp.Flag("use-fake-device-for-media-stream", true),
+
+		// custom args
+		chromedp.Flag("incognito", true),
+		chromedp.Flag("kiosk", true),
+		chromedp.Flag("enable-automation", false),
+		chromedp.Flag("autoplay-policy", "no-user-gesture-required"),
+		chromedp.Flag("window-position", "0,0"),
+		chromedp.Flag("window-size", fmt.Sprintf("%d,%d", cfg.Width, cfg.Height)),
+		chromedp.Flag("display", fmt.Sprintf(":%d", displayID)),
+	}
+
+	contextOpts := []chromedp.ContextOption{
+		chromedp.WithErrorf(slogDebugF),
+	}
+
+	if devMode := os.Getenv("DEV_MODE"); devMode == "true" {
+		opts = append(opts, chromedp.NoSandbox)
+		contextOpts = append(contextOpts, chromedp.WithLogf(slogDebugF))
+		contextOpts = append(contextOpts, chromedp.WithDebugf(slogDebugF))
+	}
+
+	if insecureOrigins, err := getInsecureOrigins(cfg.SiteURL); err != nil {
```

Comment:
I'm trying to understand the issue we're addressing here, and the part I don't get is how are you setting the SiteURL to point to the internal URL you want to route things to? The plugin won't know about it, right? Are you doing some kind of mapping in the offloader? How does using the wrong SiteURL affect the webapp, will it still work normally?

Reply:
I mean the Chromium instance will load `http://mm.svc.cluster.local:8065` based on the override, which will serve the webapp from that origin. But the *actual* SiteURL in the server config (that also gets applied in various parts of the webapp) is something like `https://community.mattermost.com`. Won't the mismatch cause problems with the CSP and potentially a number of other things?

I assume you've tested this, but just confirming because I don't really understand how it'll work.

Reply:
Confirmed that the MM server config is not even fetched in that scenario so the only URL used it the current one for both WebSocket connection and HTTP requests.